drivers/serial/uart_nrfx_uart.c

```diff
@@ -229,6 +228,11 @@ static void uart_nrfx_poll_out(struct device *dev,
 	 * blue moon against hanging up the whole thread permanently
 	 */
 
+#ifdef CONFIG_UART_0_ASYNC
```

Comment:
I don't think this fixes the core problem, which is that the condition listed in the comment is not actually applied.
'The UART API dictates that poll_out should wait for the transmitter to be empty before sending a character'
Given that UART doesn't have the TXSTARTED/TXSTOPPED events, perhaps polling until the TXDRDY interrupt is not enabled would work?

Reply:
Problem here was that event was cleared in ISR instead of poll_out function. Condition in comment is applied and working.

Reply:
why not use same mechanism for business detection as in async call?
`if (atomic_cas(&txbuf, NULL, &c) == false) return;` to just skip poll out when async transfer is ongoing and make driver busy for that time. 

Evantually, we will convert to tty (console and logger) so dropping bytes on conflict should be accepted.